src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/ConfigureGeneratedCodeAnalysisFix.cs

```diff
@@ -0,0 +1,99 @@
+                        CodeAnalysisDiagnosticsResources.ConfigureGeneratedCodeAnalysisFix,
+                        cancellationToken => ConfigureGeneratedCodeAnalysisAsync(context.Document, diagnostic.Location.SourceSpan, cancellationToken),
+                        equivalenceKey: nameof(ConfigureGeneratedCodeAnalysisFix)),
+                    diagnostic);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private async Task<Document> ConfigureGeneratedCodeAnalysisAsync(Document document, TextSpan sourceSpan, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var generatedCodeAnalysisFlags = semanticModel.Compilation.GetTypeByMetadataName(typeof(GeneratedCodeAnalysisFlags).FullName);
+            if (generatedCodeAnalysisFlags is null)
+            {
+                return document;
+            }
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var analysisContextParameter = root.FindNode(sourceSpan, getInnermostNodeForTie: true);
+
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var parameterDeclaration = analysisContextParameter;
+            var declarationKind = generator.GetDeclarationKind(parameterDeclaration);
+            while (declarationKind != DeclarationKind.Parameter)
+            {
+                parameterDeclaration = generator.GetDeclaration(parameterDeclaration.Parent);
+                if (parameterDeclaration is null)
+                {
+                    return document;
+                }
+
+                declarationKind = generator.GetDeclarationKind(parameterDeclaration);
+            }
+
+            var methodDeclaration = parameterDeclaration.Parent;
+            declarationKind = generator.GetDeclarationKind(methodDeclaration);
+            while (declarationKind != DeclarationKind.Method)
+            {
+                methodDeclaration = generator.GetDeclaration(methodDeclaration.Parent);
+                if (methodDeclaration is null)
+                {
+                    return document;
+                }
+
+                declarationKind = generator.GetDeclarationKind(methodDeclaration);
+            }
+
+            var statements = generator.GetStatements(methodDeclaration);
+            var newInvocation = generator.InvocationExpression(
+                generator.MemberAccessExpression(
+                    generator.IdentifierName(generator.GetName(parameterDeclaration)),
+                    nameof(AnalysisContext.ConfigureGeneratedCodeAnalysis)),
+                generator.BitwiseOrExpression(
+                    generator.MemberAccessExpression(
+                        generator.TypeExpressionForStaticMemberAccess(generatedCodeAnalysisFlags),
+                        nameof(GeneratedCodeAnalysisFlags.Analyze)),
```

Comment:
Wouldn't the most common case be `GeneratedCodeAnalysisFlags.None`? Should we offer multiple code fixes, one each for single bit set and then one for `Analyze | ReporDiagnostics`?

Reply:
> Wouldn't the most common case be GeneratedCodeAnalysisFlags.None?

It could be, but for a user not familiar with the API I don't think it's obvious that `None` would be a behavior change. I opted for the default since 1) it's safe to apply without code breaking and 2) the enum member names make it clear what the call is going to do.